src/DICOM.jl

```diff
@@ -176,7 +176,7 @@ function pixeldata_parse(st, sz, vr, dcm)
         read!(st, data)
     else
         # start with Basic Offset Table Item
-        data = [element(st, false)]
+        data = Array{Any,1}(element(st, false))
```

Comment:
You’re welcome. It hope there are many more to come ;) And thanks for the advice on code style.

I think it would be good to have a closer look at the tests next, as you suggested. I didn’t yet look at the specific images that Rene uses in runtests.jl but I see different behavior locally when using different files, obviously because of the many degrees of freedom of a .dcm file.
One particular thing that I noticed is that if the value representation in the header is explicit but for the content it is implicit, as defined in the following header field:

(0002,0010) UI =LittleEndianImplicit                    #  18, 1 TransferSyntaxUID

then the file won’t load (it will read through until the end and throw an end-of-file error).

Unrelated question: If I install the package using Pkg.add(“DICOM”) I still get the old version with all the deprecation warnings. I assume that’s because you have to make a release first, right? Are we aiming for another release once the tests are set up?

Cheers,
Frank







> On Dec 15, 2015, at 12:24 AM, Isaiah <notifications@github.com> wrote:
> 
> In src/DICOM.jl <https://github.com/ihnorton/DICOM.jl/pull/11#discussion_r47597832>:
> 
> > @@ -176,7 +176,7 @@ function pixeldata_parse(st, sz, vr, dcm)
> >          read!(st, data)
> >      else
> >          # start with Basic Offset Table Item
> > -        data = [element(st, false)]
> > +        data = Array{Any,1}(element(st, false))
> vector with type specification can also be written as Any[...], depending on personal aesthetics.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/ihnorton/DICOM.jl/pull/11/files#r47597832>.
>